File: Structural/Composite/Composite_Java/Component.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Composite_Java;

import java.util.ArrayList;

/**
 *
 * @author shananu1395
 */
interface Component {
    public void showPrice();
}

class Leaf implements Component{
    int price;
    String name;

    public Leaf(int price, String name) {
        this.price = price;
        this.name = name;
    }
    
    @Override
    public void showPrice() {
        System.out.println(name+" : "+price);
    }

}

class Composite implements Component{
    String name;

    public Composite(String name) {
        this.name = name;
    }
    
    ArrayList<Component> component=new ArrayList<Component>();
    @Override
    public void showPrice() {
        System.out.println(name);
        for (Component component1 : component) {
            component1.showPrice();
       }
    }
    
    public void addComponent(Component comp){
        component.add(comp);
    }

}
 

File: Structural/Composite/Composite_py/Component.py
import abc

class Component(metaclass=abc.ABCMeta):
    @abc.abstractmethod
    def showPrice(self):
        pass

class Leaf(Component):
    def __init__(self,cost,name):
        self.cost=cost
        self.name=name
    
    def showPrice(self):
        print(self.name+" : "+str(self.cost))

class Composite(Component):
    def __init__(self,name):
        self.name=name
        self.components=[]

    def addComponent(self,c):
        self.components.append(c)

    def showPrice(self):
        print(self.name)
        for i in self.components:
            i.showPrice()
